crypto/ledger_test.go

```diff
@@ -208,7 +208,7 @@ func TestSignaturesHD(t *testing.T) {
 	for account := uint32(0); account < 100; account += 30 {
 		msg := getFakeTx(account)
 
-		path := *hd.NewFundraiserParams(account, account/5)
+		path := *hd.NewFundraiserParams(account, sdk.CoinType, account/5)
```

Comment:
Could we have tests that check that confirm that results are different and correct when coinType changes?
I have not seen any tests checking that coinType actually does result in a different behavior apart from the previous fixed value (118)

Reply:
> This will break Ledger app compatibility. Ledger enforces BIP44 registrations and apps are signed and restricted to paths that start with 44'/118'. Atoms are registered at 118: https://github.com/satoshilabs/slips/blob/master/slip-0044.md

Any project based on cosmos-sdk should publish their own ledger app, if they want to singed by ledger.

> If derivation paths do not start with 44'/118' any cryptography operation will be rejected by the secure element. A new app will need to be published for a different coinType

Yes

Reply:
`fullHdPath` was only used by DerivePrivateKeyForPath() which was a general algorithm.

Reply:
My point is that now we have a new argument that is being passed through layers and at any point it could be possible that it is not being used properly. It would be nice to show that functionality and tests still work if the coin type is `12345` and that the results and test vectors do change.
It would be bad if in the future there are situations where this new argument is ignored and we get keys with the wrong coin type.

Reply:
Did you mean I should add a test just like the following `12345`?

```go
func TestStringifyFundraiserPathParams(t *testing.T) {
	path := NewFundraiserParams(4, types.CoinType, 22)
	require.Equal(t, "44'/118'/4'/0/22", path.String())

	path = NewFundraiserParams(4, types.CoinType, 57)
	require.Equal(t, "44'/118'/4'/0/57", path.String())

	path = NewFundraiserParams(4, 12345, 57)
	require.Equal(t, "44'/12345'/4'/0/57", path.String())
}
```

Reply:
Yes, something as simple as that would work.